pkg/tls/codec.go

```diff
@@ -48,16 +48,28 @@ func encode(buf *bytes.Buffer, blk *pem.Block) {
 
 // === DECODE ===
 
-// DecodePEMKey parses a PEM-encoded ECDSA private key from the named path.
-func DecodePEMKey(txt string) (*ecdsa.PrivateKey, error) {
+// DecodePEMKey parses a PEM-encoded private key from the named path.
+func DecodePEMKey(txt string) (GenericPrivateKey, error) {
 	block, _ := pem.Decode([]byte(txt))
 	if block == nil {
 		return nil, errors.New("Not PEM-encoded")
 	}
-	if block.Type != "EC PRIVATE KEY" {
-		return nil, fmt.Errorf("Expected 'EC PRIVATE KEY'; found: '%s'", block.Type)
+	switch block.Type {
+	case "EC PRIVATE KEY":
+		k, err := x509.ParseECPrivateKey(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		return PrivateKeyEC{k}, nil
+	case "RSA PRIVATE KEY":
+		k, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		return PrivateKeyRSA{k}, nil
+	default:
+		return nil, fmt.Errorf("Unsupported block type: '%s'", block.Type)
```

Comment:
nit: errors should not be capitalized... https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Indeed. Note to self: Logs can be capitalized, error strings not.